feat(coordinator): add endpoint to close disasters

mehedi-hasan-2302 opened the pull request:
Implemented closeDisaster service function to update the disaster status to 'Closed' and set the endDate. Added PATCH in the coordinator controller with authorization checks and error handling.

Warhammer4000 commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/warhammer4000/pr-checks/6377ec8e-1156-499e-aa24-c79ecb451b7d?refs=eb713cbcbebff5ff41da9b9dd70c0f117a47477b&source=prChecksComment)

mehedi-hasan-2302 committed:
feat(coordinator): add endpoint to close disasters

Implemented closeDisaster service function to update the disaster status to 'Closed' and set the endDate. Added PATCH in the coordinator controller with authorization checks and error handling.
